src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData)
+    } else if (action.type === 'keysToRemove') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    } else {
+      newState = {};
+    }
```

Comment:
Instead of using if-else statements, a switch statement would be more appropriate here since you are checking the value of 'action.type' against a limited set of conditions.